narrow_valley.py

```diff
@@ -0,0 +1,647 @@
+
+def coin_flip():
+	return random.choice(['heads', 'tails'])
+
+def die_roll():
+	return random.randint(1, 6)
+
+def time_print(string):
+    print(string)
+    time.sleep(2)
+
+def time_print_loop(lst):
+    for i in range(len(lst)):
+        time_print(lst[i])
+
+def time_print_img(lst):
+    for i in range(len(lst)):
+        print(lst[i])
+        time.sleep(.5)
+
+def valid_input(promt, option1, option2):
+    while True:
+        response = input(promt).lower()
+        if option1 in response:
+            return option1
+        if option2 in response:
+            return option2
+        else:
+            time_print("I don't understand.")
+    return response
+
+def contine_on():
+    while True:
+        response = input("Enter (1) to continue.\n").lower()
+        if "1" in response:
+            break
+        else:
+            time_print("I don't understand.")
+
+#<------------------------------------------------------------ Fight System ------------------------------------------------------------>    
+
+def intro_fight():    
+    time_print('You are both still, waiting for the right time to make your move, and then...\n')
+
+
+def choose_stats(items):
+    opt1 = {
+        'boss_hp': 15, 
+        'player_hp': 15,
+        'boss_name': 'Elijah'
+        }
+    opt2 = {
+        'boss_hp': 15, 
+        'player_hp': 15, 
+        'boss_name': 'Clover',
+        }
+    if items['key'] == '*Primal Command*':
+        items.update(opt1)
+    else:
+        items.update(opt2)
+    return items
+
+def play_again():
+    time_print("Would you like to play again?")
+    replay = valid_input("(1) Yes\n(2) No\n", "1", "2")
+    if replay == "1":
+        play()
+    elif replay == "2":
+        time_print("Ok, thanks for playing!")
+        lst = [
+            "",
+            " ,----.                                 ,-----. ",                        
+            "'  .-./    ,--,--.,--,--,--. ,---.     '  .-.  ',--.  ,--.,---. ,--.--. ",
+            "|  | .---.' ,-.  ||        || .-. :    |  | |  | \  `'  /| .-. :|  .--' ",
+            "'  '--'  |\ '-'  ||  |  |  |\   --.    '  '-'  '  \    / \   --.|  | ",   
+            " `------'  `--`--'`--`--`--' `----'     `-----'    `--'   `----'`--' ",
+            ""    
+            ]
+        time_print_img(lst)
+
+#<----- Fight Flow ----->
+
+def pick_who_attacks(items):    
+    result = coin_flip()
+    if result == 'heads':
+        player_turn(items)
+    elif result == 'tails':
+        boss_turn(items)
+    else:
+        pick_who_attacks(items)
+
+#<-------------------------------- Boss Functions -------------------------------->
+
+def dont_run(items):
+    lst = [
+        "I will not give up!",
+        "I'm not done yet!",
+        "That won't stop me!",
+        "I can do this!",
+        "I'm not afraid!"
+        ]
+    time_print(f'''({items['player_name']}) "{random.choice(lst)}"\n''')
```

Comment:
Just for my own curiosity, what does this line of code do? I haven't seen anything like this before.

Reply:
It prints an F string that's the player's name, a random string from the list "lst" and then does a line break. I used triple-quotes to wrap it because I had already used single-quotes and double-quotes inside the F string.

Reply:
I see, thanks!